### PR TITLE
aries-vcx: Revocations

### PR DESCRIPTION
**Pull request by mirgee**
Enables more tests, mainly related to revocations + small code cleanup.

Signed-off-by: Miroslav Kovar <miroslavkovar@protonmail.com>

### CODE REVIEW

File: manage [hunk 1]
```diff
@@ -220,41 +220,41 @@ function initEnv() {
...
+#         -t "${agent}-agent-backchannel" \
+#         -f "${BACKCHANNEL_FOLDER}/Dockerfile.${agent}" "aries-backchannels/"; then
+#           echo "Docker image build failed."
+#           exit 1
+#       fi
+#     else
+#       echo "Unable to find Dockerfile to build agent: ${agent}"
+#       echo "Must be one one of: ${VALID_AGENTS}"
+#     fi
+#   done
```
Comment (swcurran) [inline manage]: I'm guessing that this change is not intentional?  Did you mean to comment out the building process?

I would like it to not default to building all agents when you don't specify at least one...

File: manage [hunk 2]
```diff
@@ -777,10 +777,16 @@ runTests() {
...
+      ${terminalEmu} docker run ${INTERACTIVE} --rm --network="host" -v ${BEHAVE_INI_TMP}:/aries-test-harness/behave.ini -v "$(pwd)/aries-test-harness/allure/allure-results:/aries-test-harness/allure/allure-results/" aries-test-harness -k ${runArgs} -f allure_behave.formatter:AllureFormatter -o ./allure/allure-results -f progress -D Acme=http://host.docker.internal:9020 -D Bob=http://host.docker.internal:9030 -D Faber=http://host.docker.internal:9040 -D Mallory=http://host.docker.internal:9050
   else
-      ${terminalEmu} docker run ${INTERACTIVE} --rm --network="host" -v ${BEHAVE_INI_TMP}:/aries-test-harness/behave.ini aries-test-harness -k ${runArgs} -D Acme=http://0.0.0.0:9020 -D Bob=http://0.0.0.0:9030 -D Faber=http://0.0.0.0:9040 -D Mallory=http://0.0.0.0:9050
+      ${terminalEmu} docker run ${INTERACTIVE} --rm --network="host" -v ${BEHAVE_INI_TMP}:/aries-test-harness/behave.ini aries-test-harness -k ${runArgs} -D Acme=http://host.docker.internal:9020 -D Bob=http://host.docker.internal:9030 -D Faber=http://host.docker.internal:9040 -D Mallory=http://host.docker.internal:9050
   fi
+  # if [[ "${REPORT}" = "allure" ]]; then
```
Comment (swcurran) [inline manage]: Should this remain here or be removed?

File: manage [hunk 3]
```diff
@@ -777,10 +777,16 @@ runTests() {
...
+      ${terminalEmu} docker run ${INTERACTIVE} --rm --network="host" -v ${BEHAVE_INI_TMP}:/aries-test-harness/behave.ini -v "$(pwd)/aries-test-harness/allure/allure-results:/aries-test-harness/allure/allure-results/" aries-test-harness -k ${runArgs} -f allure_behave.formatter:AllureFormatter -o ./allure/allure-results -f progress -D Acme=http://host.docker.internal:9020 -D Bob=http://host.docker.internal:9030 -D Faber=http://host.docker.internal:9040 -D Mallory=http://host.docker.internal:9050
   else
-      ${terminalEmu} docker run ${INTERACTIVE} --rm --network="host" -v ${BEHAVE_INI_TMP}:/aries-test-harness/behave.ini aries-test-harness -k ${runArgs} -D Acme=http://0.0.0.0:9020 -D Bob=http://0.0.0.0:9030 -D Faber=http://0.0.0.0:9040 -D Mallory=http://0.0.0.0:9050
+      ${terminalEmu} docker run ${INTERACTIVE} --rm --network="host" -v ${BEHAVE_INI_TMP}:/aries-test-harness/behave.ini aries-test-harness -k ${runArgs} -D Acme=http://host.docker.internal:9020 -D Bob=http://host.docker.internal:9030 -D Faber=http://host.docker.internal:9040 -D Mallory=http://host.docker.internal:9050
```
Comment (swcurran) [inline manage]: @WadeBarnes -- does this change look good to you?  You are way deeper into the cross-platform issues with docker than I. :-).

File: .github/workflows/test-harness-ariesvcx-ariesvcx.yml [hunk 4]
```diff
@@ -41,7 +37,7 @@ jobs:
         with:
           BUILD_AGENTS: "-a aries-vcx"
           TEST_AGENTS: "-d aries-vcx"
-          TEST_SCOPE: "-t @T001-RFC0036,@T002-RFC0036,@T003-RFC0036,@T001-RFC0037,@T001.2-RFC0037,@T001.3-RFC0037,@T001.4-RFC0037,@T001.5-RFC0037,@T003-RFC0037,@T003.1-RFC0037,@T001-RFC0160,@T002-RFC0160,@T003-RFC0160"
+          TEST_SCOPE: "-t @RFC0036,@RFC0037,@RFC0160,@revocation -t ~@wip -t ~@DIDExchangeConnection -t ~@T002-HIPE0011"
```
Comment (swcurran) [inline .github/workflows/test-harness-ariesvcx-ariesvcx.yml:40]: Should the test scope change with VCX to other frameworks -- e.g. VCX to ACA-Py?

File: manage (hunk 3 above)
Comment (swcurran) [reply manage]: I had to undo all the changes to get the script to work on Ubuntu - none of the containers started, neither the backchannel or the services.

**Review by swcurran — CHANGES_REQUESTED** [review]
Other than backing out the `./manage` change, things are looking good:

```
3 features passed, 1 failed, 4 skipped
40 scenarios passed, 1 failed, 69 skipped
367 steps passed, 1 failed, 627 skipped, 0 undefined
Took 14m5.272s
```

About 112 more steps passed then the same set of tests before the PR.  Yay!!

I did notice that the `agency` and `postgres` containers are not being stopped, but I thought I would add an issue for that.

I'm going to be running an ACA-Py - VCX run as well.  For now, as long as the ./manage scripts can be addressed, we can merge this.

Nice work!

File: manage (hunk 1 above)
Comment (mirgee) [reply manage]: These changes were commited by mistake, apologies :)

File: manage (hunk 3 above)
Comment (mirgee) [reply manage]: I commited this by mistake - fixed now

File: manage (hunk 2 above)
Comment (mirgee) [reply manage]: dtto

**Review by swcurran — COMMENTED** [review]
Struggled with the ACA-Py / Aries VCX.  I tried using the same set of tags as the Aries VCX to Aries VCX, and we didn't get too many more working.

I'm happy to merge, if you would like -- let me know. But I suspect there will be a bit more to do somewhere to get more tests running.

**Comment by mirgee** [comment]
@swcurran It seems acapy does not receive credential offer sent from aries-vcx - trying to figure out why...

**Comment by mirgee** [comment]
@swcurran It seems that when aries-vcx sends credential offer, acapy seems to receive it, but goes to `Complete` state which the harness doesn't expect and so times out waiting for the correct state. I am not sure what is causing it - the message might be in a format acapy doesn't expect and so it fails silently, but the `acapy::record::issue_credential::offer_received` event is still sent 🤷‍♂️

**Comment by swcurran** [comment]
@ianco -- can you please help investigate on the ACA-Py side of this?  How is ACA-Py handling credential offers from Aries VCX?  See the note above and perhaps ask really good questions to help us get to the bottom of this.

**Comment by ianco** [comment]
I get the following results with this run:

`./manage run -d acapy-main -b aries-vcx -t @RFC0036,@RFC0037,@RFC0160,@revocation -t ~@wip -t ~@DIDExchangeConnection -t ~@T002-HIPE0011`

... results in:

```
Failing scenarios:
  features/0160-connection.feature:52  Inviter Sends invitation for one agent second agent tries after connection
  features/0160-connection.feature:69  Inviter Sends invitation for one agent second agent tries during first share phase
  features/0183-revocation.feature:24  Credential revoked by Issuer and Holder attempts to prove with a prover that doesn't care if it was revoked -- @1.1 
  features/0183-revocation.feature:60  Credential revoked by Issuer and Holder attempts to prove with a prover that doesn't care if it was revoked -- @1.1 
  features/0183-revocation.feature:116  Proof in process while Issuer revokes credential before presentation and the verifier doesn't care about revocation status -- @1.1 
  features/0183-revocation.feature:155  Credential is revoked inside the non-revocation interval -- @1.1 
  features/0183-revocation.feature:156  Credential is revoked inside the non-revocation interval -- @1.2 
  features/0183-revocation.feature:157  Credential is revoked inside the non-revocation interval -- @1.3 
  features/0183-revocation.feature:176  Credential is revoked before the non-revocation instant -- @1.1 
  features/0183-revocation.feature:177  Credential is revoked before the non-revocation instant -- @1.2 
  features/0183-revocation.feature:178  Credential is revoked before the non-revocation instant -- @1.3 
  features/0183-revocation.feature:179  Credential is revoked before the non-revocation instant -- @1.4 
  features/0183-revocation.feature:197  Credential is revoked before the non-revocation interval -- @1.1 
  features/0183-revocation.feature:198  Credential is revoked before the non-revocation interval -- @1.2 
  features/0183-revocation.feature:216  Credential is revoked before the non-revocation instant -- @1.1 
  features/0183-revocation.feature:217  Credential is revoked before the non-revocation instant -- @1.2 
  features/0183-revocation.feature:218  Credential is revoked before the non-revocation instant -- @1.3 
  features/0183-revocation.feature:219  Credential is revoked before the non-revocation instant -- @1.4 
  features/0183-revocation.feature:333  Revocable Credential not revoked and Holder attempts to prove without a non-revocation interval -- @1.1 
  features/0183-revocation.feature:350  Non-revocable Credential, not revoked, and holder proves claims with the credential with timesstamp -- @1.1 
  features/0183-revocation.feature:367  Revocable Credential, not revoked, and holder proves claims with the credential with timesstamp -- @1.1 

2 features passed, 2 failed, 4 skipped
20 scenarios passed, 21 failed, 69 skipped
237 steps passed, 21 failed, 737 skipped, 0 undefined
```

**Comment by swcurran** [comment]
I think we should merge and go from there.  Everyone OK with that?

**Review by swcurran — APPROVED** [review]
I think there is still work to be done an ACA-Py (and others) to Aries VCX, but we've got some good progress here that is worth merging.

**Comment by ianco** [comment]
> I think we should merge and go from there. Everyone OK with that?

Works for me

**Pull request merged**